### DISCUSSION TITLE
Is it possible to get compile time / linter check for providing all required fields during an entity creation?

### DISCUSSION THREAD

**Discussion by vamshiaruru-virgodesigns** [ent/ent#3502] [Q&A] [answered]
Hi everyone, I am trying out ent for the first time, and I am liking what I am seeing so far. I have a couple of questions. Firstly, if I forget to provide a field using the `.Set<FieldName>` in the builder, it fails at run time. For a entity with lot of fields, it is likely that I might forget setting a field. Is it possible to get this feedback immediately instead of waiting until run time? I use golangclint with exhaustruct to make sure all fields in a struct are provided, and this is good enough in sqlc. Is there something I can do in ent as well, to get the error immediately?

Secondly, I'm following the example provided in the documentation. For this function:

```go
// Save creates the User in the database.
func (uc *UserCreate) Save(ctx context.Context) (*User, error) {
	return withHooks[*User, UserMutation](ctx, uc.sqlSave, uc.mutation, uc.hooks)
}
```

I am getting the following warning from vscode: 
```
unnecessary type arguments infertypeargs
```
Is this a known issue? I don't want to edit an auto generated file, so how do I fix this? 

Thirdly, is there any way to print what query ent is executing without actually executing it? A dry run or demo mode of sorts?

Thanks for the help!

**Comment by a8m** [comment, marked as answer]
Hey, @vamshiaruru-virgodesigns 👋 
Sharing my answer on [Discord](https://discord.com/channels/885059418646003782/885059419107360779/1100338153648312320).
